client/cgo/verification.go

```diff
@@ -112,8 +142,15 @@ func cgoVerifyAuthPath(treeHash unsafe.Pointer, treeHashSize C.int,
 	ap.TreeNonce = tn
 	ap.LookupIndex = li
 	ap.PrunedTree = pt
+	ap.Leaf = &leafNode{
```

Comment:
I got that defining the leaf in the client is what simplifies the verification function signatures. I was confused about your statement that doubling down on `nil`s makes the signature simpler. Anyways, I didn't realize the `leafNode` you've set in the client is just a placeholder. The client functions will certainly evolve as we flesh out the messages. 

> eventually cgoVerifySignature, cgoVerifyCommitment, will be combined in more elaborate cgoDoTheClientBusiness type thing.

Yeah, sounds about right.